hcache/hcache.c

```diff
@@ -378,11 +473,24 @@ void *mutt_hcache_fetch_raw(header_cache_t *hc, const char *key, size_t keylen)
     return NULL;
 
   struct Buffer path = mutt_buffer_make(1024);
-
+  size_t dlen;
   keylen = mutt_buffer_printf(&path, "%s%s", hc->folder, key);
-
-  void *blob = ops->fetch(hc->ctx, mutt_b2s(&path), keylen);
+  void *blob = ops->fetch(hc->ctx, mutt_b2s(&path), keylen, &dlen);
   mutt_buffer_dealloc(&path);
+
+#ifdef USE_HCACHE_COMPRESSION
+  if (C_HeaderCacheCompressMethod)
+  {
+    if (blob == NULL)
+      return blob;
+
+    const struct ComprOps *cops = compr_get_ops();
+    hc->ondisk = blob;
+
+    return cops->decompress(hc->cctx, blob, dlen);
+  }
+#endif
```

Comment:
... and I also suggest to simplify the conditional to set `blob`.

```c
if (C_HeaderCacheCompressMethod)
{
  if (blob != NULL)
  {
    const struct ComprOps *cops = compr_get_ops();
    hc->ondisk = blob;
    blob = cops->decompress(hc->cctx, blob, dlen);
  }
}
```

Reply:
Done, first `if()` is like this now:
`if (C_HeaderCacheCompressMethod && blob != NULL)`

Reply:
And blob is set...